transforms/deprecate-create/deprecate-create.js

```diff
@@ -0,0 +1,143 @@
+/*
+ * Convert all <Create /> to
+ * <Button buttonType=“dashed” iconType=“plus” iconPosition=“after” fullWidth />
+ */
+import registerMethods from "../registerMethods";
+
+function transformer(fileInfo, api, options) {
+  const j = api.jscodeshift;
+  registerMethods(j);
+
+  const newProps = [
+    j.jsxAttribute(j.jsxIdentifier("buttonType"), j.literal("dashed")),
+    j.jsxAttribute(j.jsxIdentifier("iconType"), j.literal("plus")),
+    j.jsxAttribute(j.jsxIdentifier("iconPosition"), j.literal("after")),
+    j.jsxAttribute(j.jsxIdentifier("fullWidth")),
+  ];
+
+  /*
+    Change import from 'import Create from "carbon-react/lib/components/create";'
+    to 'import Button from "carbon-react/lib/components/button";'
+  */
+  const ImportReplacement = () => {
+    const createImport = root.find(j.ImportDeclaration, {
+      source: {
+        type: "Literal",
+        value: "carbon-react/lib/components/create",
+      },
+    });
+
+    const buttonImport = root.find(j.ImportDeclaration, {
+      source: {
+        type: "Literal",
+        value: "carbon-react/lib/components/button",
+      },
+    });
+
+    // If there's already a Button import, just remove the Create one
+    if (buttonImport.size()) {
+      createImport.remove();
+      return true;
+    }
+
+    if (createImport.size()) {
+      createImport.replaceWith(
+        j.importDeclaration(
+          [j.importDefaultSpecifier(j.identifier("Button"))],
+          j.literal("carbon-react/lib/components/button")
+        )
+      );
+      return true;
+    }
+  };
+
+  const OpeningTagReplacement = (tag) => {
```

Comment:
We can simplify this by spreading the new props and any existing props into a new array. Theres no difference between the actual replacements so we can remove the if/else.

```suggestion
const OpeningTagReplacement = (tag) => {
    const name = tag.find(j.JSXIdentifier, { name: "Create" });
    if (name.size()) {
      const currentAttributes = tag.find(j.JSXAttribute);

      let props = [
        ...currentAttributes.nodes(),
        j.jsxAttribute(j.jsxIdentifier("buttonType"), j.literal("dashed")),
        j.jsxAttribute(j.jsxIdentifier("iconType"), j.literal("plus")),
        j.jsxAttribute(j.jsxIdentifier("iconPosition"), j.literal("after")),
        j.jsxAttribute(j.jsxIdentifier("fullWidth")),
      ];

      const selfClosing = tag.nodes()[0].selfClosing;
      tag.replaceWith(
        j.jsxOpeningElement(j.jsxIdentifier("Button"), props, selfClosing)
      );
      return true;
    }
  };
```

Reply:
Ah this is exactly what I wanted to do but I couldn't work out how - must have missed the nodes() function in the docs

Reply:
Done